applications/ortege.md

```diff
@@ -0,0 +1,267 @@
+
+We have also made substantial progress on the smart contract front. Our team has deployed smart contracts onto EVM chains and achieved key milestones, including the successful transmission of interchain messages, the creation of a bridge for tokens, and the ability to remotely read state on smart contracts. These accomplishments demonstrate our technical prowess and the functional capabilities of our protocol. 
+
+To provide transparency and traceability, we have deployed an explorer that can track the entire lifecycle of a message. This tool allows users to monitor the status of their transactions and provides valuable insights into the operation of the protocol. 
+
+Lastly, we have integrated WalletConnect 2 to enable the remote control of an interchain account on another chain. This feature enhances the user experience by allowing users to manage their assets across different chains from a single interface. 
+
+In summary, while there is still work to be done, we are proud of the progress we have made so far and are excited about the potential of Ortege to revolutionize blockchain interoperability. 
+
+## Development Roadmap :nut_and_bolt: 
+ ### Overview 
+- **Total Estimated Duration:** 5 months 
+
+- **Full-Time Equivalent (FTE):**  4 FTE (Milestones will be worked on concurrently) 
+
+- **Total Costs:** $95,000 
+
+  
+
+### Milestone 1 Ink! Smart Contracts 
+
+We will begin by developing smart contracts in the ink! language. These contracts will form the backbone of our protocol, enabling the core functionality of cross-chain communication. We will leverage the design principles and specifications provided by Hyperlane to ensure our contracts are robust, secure, and efficient. This will involve creating contracts for managing the protocol's state, handling message routing, and facilitating the transfer of value across chains. 
+ 
+- **Estimated duration:** 4 months 
+- **FTE:**  2.5 
+- **Costs:** 60,000 USD 
+
+| Number | Deliverable | Specification | 
+| ------ | ----------- | ------------- | 
+| **0a.** | License | Apache 2.0 | 
+| **0b.** | Documentation | We will provide both **inline documentation** of the code and a basic **tutorial** that explains how a user can query, call or post an interchain message. | 
+| **0c.** | Testing and Testing Guide | Core functions will be fully covered by comprehensive unit tests to ensure functionality and robustness. In the guide, we will describe how to run these tests. | 
+| **0d.** | Docker | We will provide a Dockerfile(s) that can be used to test all the functionality delivered with this milestone. | 
+| **0e.** | Article | We will publish an **article**/workshop that explains [...] (what was done/achieved as part of the grant). | 
+| **1.** | Ink! Contract: MerkleTree | We will create an ink! MerkleTree implementation and will submit to OpenBrush for inclusion in their standards | 
```

Comment:
What kind of merkle tree will you implement? For example, a patricia merkle tree? Or a MMR? Or a binary tree? What is it going to be used for? Will the by any different to the existing implementations, e.g.:
- [binary merkle tree](https://github.com/paritytech/substrate/blob/master/utils/binary-merkle-tree/src/lib.rs)
- [MMR](https://github.com/paritytech/substrate/blob/master/primitives/merkle-mountain-range/src/lib.rs)
- see also the [solidity-merkle-trees](https://github.com/polytope-labs/solidity-merkle-trees/tree/main) repo

Reply:
It's an incremental Merkle Tree as can be seen here https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/401b32ddddc6faa281beb050f641afb4f024d905/solidity/contracts/libs/Merkle.sol#L9

Reply:
It's because it's what Hyperlane use, so we want to be consistent

Reply:
Here's a rust-based implementation of an incremental merkle tree: https://docs.rs/incrementalmerkletree/latest/incrementalmerkletree/

Is it possible to reuse this one?